htmresearch/frameworks/nlp/classify_document_fingerprint.py

```diff
@@ -204,35 +199,46 @@ def trainText(self, token, labels, sequenceId=None, reset=0):
     if reset == 1:
       document = " ".join(self.currentDocument)
       sensor = self.sensorRegion.getSelf()
-      sensor.addDataToQueue(document, labels, sequenceId, reset)
+      sensor.addDataToQueue(document, labels, sampleId, reset)
 
       for region in self.learningRegions:
         region.setParameter("learningMode", True)
       self.network.run(1)
-
+      self.reset()
       self.currentDocument = None
 
       # Print the outputs of each region
       if self.verbosity >= 2:
         print "Training with document:",document
-        print "SequenceId:",sequenceId
+        print "SequenceId:",sampleId
         if self.verbosity >= 3:
           self.printRegionOutputs()
 
 
-  def classifyText(self, token, reset=0):
+  def inferToken(self, token, reset=0, returnDetailedResults=False,
+                 sortResults=True):
     """
-    Classify the token and return a list of the best classifications.
+    Classify the token (i.e. run inference on the model with this document) and
+    return classification results and a list of sampleIds and distances.
+    Repeated sampleIds are NOT removed from the results.
 
     @param token    (str)  The text token to train on
     @param reset    (int)  Should be 0 or 1. If 1, assumes we are at the
                            end of a sequence. A reset signal will be issued
                            after the model has been trained on this token.
+    @param returnDetailedResults
+                    (bool)    If True will return sampleIds and distances
+                              This could slow things down depending on the
+                              number of stored patterns.
+    @param sortResults (bool) If true the list of sampleIds and distances
+                              will be sorted in order of increasing distances.
 
     @return  (numpy array) An array of size numLabels. Position i contains
-                           the likelihood that this sample belongs to the
+                           the likelihood that this token belongs to the
```

Comment:
an array of all zeros implies we have not yet run inference b/c the reset for end-of-document has not been flagged